src/hooks/useSearch.ts

```diff
@@ -0,0 +1,85 @@
+import { useState } from 'react';
+import { WikiWord } from '../types';
+import KOREAN_CONSONANTS from '../../static/korean-consonants';
+
+const UNICODE_FIRST_FOREAN = 44032;
+const NUMBER_OF_KOREAN_LETTERS = 11172;
+const NUMBER_OF_KOREAN_NUCLEUS = 21;
+const NUMBER_OF_KOREAN_CODA = 28;
+
+const ALPHABET_REGEX = new RegExp(/^[A-Za-z]$/);
+const KOREAN_CONSONANTS_REGEX = new RegExp(/^[ㄱ-ㅎ]$/);
+
+const isLetterFilterValue = (value: string): boolean => {
+  return ALPHABET_REGEX.test(value) || KOREAN_CONSONANTS_REGEX.test(value);
+};
+const getLetterLanguage = (value: string): string => {
+  return KOREAN_CONSONANTS_REGEX.test(value)
+    ? 'KOREAN'
+    : ALPHABET_REGEX.test(value)
+    ? 'ENGLISH'
+    : '';
+};
```

Comment:
```suggestion
const getLetterLanguage = (value: string): 'KOREAN' | 'ENGLISH' | null => {
  if (KOREAN_CONSONANTS_REGEX.test(value)) {
    return 'KOREAN';
  }
  if (ALPHABET_REGEX.test(value)) {
    return 'ENGLISH';
  }
  return null;
};
```

1. Union Type 이용해서 return type 더 정확하게 줄 수 있을 것 같아요!
2. 3항연산자를 제거해 가독성 향상

Reply:
위에 삼항연산자 부분 어떻게 고칠까 생각했었는데 좋은 힌트 주셔서 감사합니다!! 적용해 보겠숩니당!

Reply:
위의 3항연산자 부분 어떻게 할까 고민하고 있었습니다. 힌트 주셔서 감사합니당! 
더 정확하고 깔끔한 코드로 보입니당!! 반영하겠숩니당!

Reply:
저도 삼항연산자 굉장히 좋아하는데, 저런식으로 중첩해서 써야 하는경우는 @greatSumini 님 말씀처럼 귀찮더라도 풀어서 쓰는 편입니다.
더불어
```
  if (KOREAN_CONSONANTS_REGEX.test(value)) {
    return 'KOREAN';
  }
  if (ALPHABET_REGEX.test(value)) {
    return 'ENGLISH';
  }
```
이러면 첫번째 if 랑 상관없이 if 문이 두번 실행되기 때문에
```
  if (KOREAN_CONSONANTS_REGEX.test(value)) {
    return 'KOREAN';
  }
  else if (ALPHABET_REGEX.test(value)) {
    return 'ENGLISH';
  }
```
이렇게 하는게 더 좋지 않을까 생각해요.

그리고 질문이 있는데, 만약에 한/영이 모두 쓰인 keyword는 어떻게 되나요?
혹은 이런 경우가 발생치 않나요?

Reply:
1. 그렇네요 if을 한번 실행하는 부분 너무 좋아요!
2. 한/영 둘다 쓰인 키워드일 경우에는 저희 용어집 정규식 " 한글 (영어) " 에 따라서  충분히 필터할 수 있을 거라고 생각하고 구현을 했습니다. 
만약 한글 영어가 섞여있고 첫 글자가 영어라면 한글 필터에서는 필터 되지 않는게 좋다고 생각합니다. 
ex) "GA의 데이터 구조" 라면 첫 글자 영어이기에 영어에서만 걸러집니다. 첫 글자를 특정하기 힘들다고 생각했습니다. 한글의 첫자라면 "의" 인데 o을 눌렀을 때 나오는 것이 유의미하다고 생각하지 않았습니당! 이런 경우 그냥 "G" 에서만 필터링 되는게 좋다고 생각했습니다!